File: rust/src/types/http_client.rs

trait HttpClient {
	fn CloseIdleConnections();
	fn Do(req: hyper::Request<Body>) -> Result<hyper::Response<Body>, Error>;
	fn Get(url: string)->  Result<hyper::Response<Body>, Error>;
	fn Head(url: string)->  Result<hyper::Response<Body>, Error>;
	fn Post(url: string, contentType: string, body: std::io::Reader)->  Result<hyper::Response<Body>, Error>;
	// fn PostForm(url: string, data: url.Values)->  (resp *hyper::Response<Body>, err error)
}

File: ts/src/types/http_client.ts
import { HttpRequest } from "./http_request";
import { HttpResponse } from "./http_response";

export interface HttpClient {
  // CloseIdleConnections()
  Do(req: HttpRequest): Promise<HttpResponse>;
  Get(url: string): Promise<HttpResponse>;
  // Head(url string) (resp *http.Response, err error)
  Post(url: string, contentType: string, body: ReadableStream<Uint8Array> | Uint8Array | string): Promise<HttpResponse>;
  // PostForm(url string, data url.Values) (resp *http.Response, err error)
}
